server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java

```diff
@@ -355,7 +355,7 @@ protected void shardOperation(final UpdateRequest request, final ActionListener<
                 DocStatusStats stats = new DocStatusStats();
                 stats.inc(RestStatus.OK);
 
-                indicesService.addDocStatusStats(stats);
+                indicesService.addNewDocStatusStats(stats);
```

Comment:
All of these changes can be avoided when you change the existing method instead.